packages/opentelemetry-exporter-prometheus/src/PrometheusSerializer.ts

```diff
@@ -0,0 +1,250 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { MetricRecord, AggregatorKind } from '@opentelemetry/metrics';
+import { PrometheusExportType, PrometheusCheckpoint } from './types';
+import { Labels } from '@opentelemetry/api';
+import { hrTimeToMilliseconds } from '@opentelemetry/core';
+
+function escapeString(str: string) {
+  return str.replace(/\n/g, '\\n').replace(/\\(?!n)/g, '\\\\');
+}
+
+function escapeLabelValue(str: string) {
+  if (typeof str !== 'string') {
+    str = String(str);
+  }
+  return escapeString(str).replace(/"/g, '\\"');
+}
+
+const invalidCharacterRegex = /[^a-z0-9_]/gi;
+/**
+ * Ensures metric names are valid Prometheus metric names by removing
+ * characters allowed by OpenTelemetry but disallowed by Prometheus.
+ *
+ * https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels
+ *
+ * 1. Names must match `[a-zA-Z_:][a-zA-Z0-9_:]*`
+ *
+ * 2. Colons are reserved for user defined recording rules.
+ * They should not be used by exporters or direct instrumentation.
+ *
+ * OpenTelemetry metric names are already validated in the Meter when they are created,
+ * and they match the format `[a-zA-Z][a-zA-Z0-9_.\-]*` which is very close to a valid
+ * prometheus metric name, so we only need to strip characters valid in OpenTelemetry
+ * but not valid in prometheus and replace them with '_'.
+ *
+ * @param name name to be sanitized
+ */
+function sanitizePrometheusMetricName(name: string): string {
+  return name.replace(invalidCharacterRegex, '_'); // replace all invalid characters with '_'
+}
+
+function valueString(value: number) {
+  if (Number.isNaN(value)) {
+    return 'Nan';
+  } else if (!Number.isFinite(value)) {
+    if (value < 0) {
+      return '-Inf';
+    } else {
+      return '+Inf';
+    }
+  } else {
+    return `${value}`;
+  }
+}
+
+function aggregatorKindToPrometheusType(
```

Comment:
please convert return types into enums

Reply:
By saying `enums`, do you suggest using TypeScript enum keyword type here?

Reply:
yes, but I see you created a types for that, we usually do mapping in such cases from one enum to another.
`[AggregatorKind.SUM]: PrometheusExportType.COUNTER`
etc.

Reply:
The reason I prefer plain strings rather than using typescript enums here is that we don't expose the return value publicly. This is a serialization method, the value returned here will be concatenated to a result string. So I believe these will be fewer benefits to using enums here.

Reply:
The type `PrometheusExportType` been exported publicly is a mistake. I'll make it private.